[2025-01-16T13:17:00.586+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: spark_parallel_processing.spark_job manual__2025-01-16T13:16:58.937688+00:00 [queued]>
[2025-01-16T13:17:00.593+0000] {taskinstance.py:1159} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: spark_parallel_processing.spark_job manual__2025-01-16T13:16:58.937688+00:00 [queued]>
[2025-01-16T13:17:00.593+0000] {taskinstance.py:1361} INFO - Starting attempt 1 of 1
[2025-01-16T13:17:00.612+0000] {taskinstance.py:1382} INFO - Executing <Task(SparkSubmitOperator): spark_job> on 2025-01-16 13:16:58.937688+00:00
[2025-01-16T13:17:00.622+0000] {standard_task_runner.py:57} INFO - Started process 94 to run task
[2025-01-16T13:17:00.626+0000] {standard_task_runner.py:84} INFO - Running: ['***', 'tasks', 'run', 'spark_parallel_processing', 'spark_job', 'manual__2025-01-16T13:16:58.937688+00:00', '--job-id', '10', '--raw', '--subdir', 'DAGS_FOLDER/dag-spark-submit.py', '--cfg-path', '/tmp/tmptcsatpip']
[2025-01-16T13:17:00.628+0000] {standard_task_runner.py:85} INFO - Job 10: Subtask spark_job
[2025-01-16T13:17:00.685+0000] {task_command.py:416} INFO - Running <TaskInstance: spark_parallel_processing.spark_job manual__2025-01-16T13:16:58.937688+00:00 [running]> on host 3c782527ad0c
[2025-01-16T13:17:00.754+0000] {taskinstance.py:1662} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='spark_parallel_processing' AIRFLOW_CTX_TASK_ID='spark_job' AIRFLOW_CTX_EXECUTION_DATE='2025-01-16T13:16:58.937688+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='manual__2025-01-16T13:16:58.937688+00:00'
[2025-01-16T13:17:00.764+0000] {base.py:73} INFO - Using connection ID 'spark_default' for task execution.
[2025-01-16T13:17:00.764+0000] {spark_submit.py:340} INFO - Spark-Submit cmd: spark-submit --master spark-master:7077 --name arrow-spark --verbose /opt/spark-apps/spark_app.py
[2025-01-16T13:17:00.864+0000] {spark_submit.py:491} INFO - /home/***/.local/lib/python3.8/site-packages/pyspark/bin/load-spark-env.sh: line 68: ps: command not found
[2025-01-16T13:17:00.866+0000] {spark_submit.py:491} INFO - JAVA_HOME is not set
[2025-01-16T13:17:00.876+0000] {taskinstance.py:1937} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/operators/spark_submit.py", line 156, in execute
    self._hook.submit(self._application)
  File "/home/airflow/.local/lib/python3.8/site-packages/airflow/providers/apache/spark/hooks/spark_submit.py", line 422, in submit
    raise AirflowException(
airflow.exceptions.AirflowException: Cannot execute: spark-submit --master spark-master:7077 --name arrow-spark --verbose /opt/spark-apps/spark_app.py. Error code is: 1.
[2025-01-16T13:17:00.879+0000] {taskinstance.py:1400} INFO - Marking task as FAILED. dag_id=spark_parallel_processing, task_id=spark_job, execution_date=20250116T131658, start_date=20250116T131700, end_date=20250116T131700
[2025-01-16T13:17:00.893+0000] {standard_task_runner.py:104} ERROR - Failed to execute job 10 for task spark_job (Cannot execute: spark-submit --master spark-master:7077 --name arrow-spark --verbose /opt/spark-apps/spark_app.py. Error code is: 1.; 94)
[2025-01-16T13:17:00.938+0000] {local_task_job_runner.py:228} INFO - Task exited with return code 1
[2025-01-16T13:17:00.950+0000] {taskinstance.py:2778} INFO - 0 downstream tasks scheduled from follow-on schedule check
